drivers/spi/spi_ll_stm32.c

```diff
@@ -460,6 +482,13 @@ static void spi_stm32_complete(const struct device *dev, int status)
 		LL_SPI_ClearFlag_MODF(spi);
 	}
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+	LL_SPI_ClearFlag_TXTF(spi);
+	LL_SPI_ClearFlag_OVR(spi);
+	LL_SPI_ClearFlag_EOT(spi);
+	LL_SPI_SetTransferSize(spi, 0);
```

Comment:
```suggestion	
```

Seems to cause extra interrupt when SPE is disabled that leads to SPE stalling when `fifo-enabled` is set. Moreover, the size is going to be set in `transceive` anyway. However, the `transceive` sets it only when `fifo-enabled` is set, that condition should be removed.